addons/pos_mercado_pago/models/pos_payment_method.py

```diff
@@ -0,0 +1,182 @@
+    mp_webhook_secret_key = fields.Char(string="Production webhook secret key", help='Mercado Pago customer production webhook secret key')
+
+    mp_last_message = fields.Json()
+    mp_last_payment_intend_id = fields.Char()
+
+    mp_smart_point_ids = fields.Many2one("pos.terminal", index=True, string="Terminal S/N", \
+        help="Given by Mercado Pago on connection, check correspondance with the serial number written on the device")
+    point_smart_op_mode = fields.Selection(
+        related='mp_smart_point_ids.point_smart_op_mode',
+        string='Operating mode',
+        help="""To function properly with the Odoo system, the operating mode must be point of sale.
+        Warning: For a given terminal, when you change the store or the point of sale from your Mercado Pago account,
+        the terminal automatically restore its mode to 'standalone'""",
+        readonly=False)
+    id_point_smart_store =fields.Char(
+        related='mp_smart_point_ids.id_point_smart_store',
+        string='Store ID',
+        readonly=True)
+    id_point_smart_pos = fields.Char(
+        related='mp_smart_point_ids.id_point_smart_pos',
+        string='POS ID',
+        readonly=True)
+    point_smart_store = fields.Text(
+        related='mp_smart_point_ids.point_smart_store',
+        string='Store',
+        help="Set the store on your Mercado Pago account",
+        readonly=True)
+    point_smart_pos = fields.Text(
+        related='mp_smart_point_ids.point_smart_pos',
+        string='Point of sale',
+        help="Set the point of sale on your Mercado Pago account",
+        readonly=True)
+
+    def _get_payment_terminal_selection(self):
+        return super()._get_payment_terminal_selection() + [('mercado_pago', 'Mercado Pago')]
+
+    @api.onchange('point_smart_op_mode')
+    def _onchange_point_smart_op_mode(self):
+        _logger.info('terminal mode changed')
+        # Avoid triggering at the creation
+        if self.use_payment_terminal != 'mercado_pago' or not self.mp_smart_point_ids.id_point_smart:
+            return
+        # User has changed the operatrion mode, translate the selection choosen and
+        # send a request to change the mode to Mercado Pago
+        mode = json.loads('{"operating_mode": "%s"}' % ("STANDALONE" if self.point_smart_op_mode == "standalone" else "PDV"))
+        # Whatever the Mercado Response...
+        self._call_mp("patch", f"/point/integration-api/devices/{self.mp_smart_point_ids.id_point_smart}", mode)
+        # We alert the user for consequence of the change
+        return {
+            'warning': {
+                'title': _("Warning"),
+                'message': _("Terminal in standalone mode will not work with the Odoo system") 
+                        if self.point_smart_op_mode == 'standalone' 
+                        else _("On the terminal, close and reopen the 'Caja' to apply the change"),
+            }
+        }           
```

Comment:
Good point. Indeed, if the terminal is in standalone mode, it ignores any request of payment intend. On the front end, a timeout is triggered because it receives nothing as notification.
Before the timeout, if the user click on cancel, it receives a message saying to cancel the transaction on the terminal (but this is impossible since it is in standalone mode).
After the timeout, if the user retry to send a payment request, it receives a message saying that a payment is already sent to the terminal (but the terminal doesn't have anything on it related with a payment intend since it's in standalone mode)
The only problem I see is, when user put the terminal on point of sale mode from the configuration page (then restart the terminal), the last payment intend is still sent to the terminal by Mercado Pago. Hence the user must cancel it because it will be never taken into account (you must close ther point of sale before changing a configuration)